File: api/Core/Model/Dosage.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Xcianify.Core.Model
{
    [Table("dosage")]
    public class Dosage : BaseModel
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; }

        [StringLength(255)]
        public string RegisterDate { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public int? CreatedBy { get; set; }

        public int? UpdatedBy { get; set; }
    }
} 

File: pharma-erp-3/types/dosage.ts
export interface Dosage {
  id: number
  name: string
  registerDate: string
  isActive: boolean
  createdAt: string
  updatedAt?: string
  createdBy?: number
  updatedBy?: number
}

export interface CreateDosageRequest {
  name: string
  registerDate?: string
  isActive: boolean
}

export interface UpdateDosageRequest {
  name: string
  registerDate?: string
  isActive: boolean
}

export interface DosageFilter {
  name?: string
  isActive?: boolean
  page?: number
  pageSize?: number
  sortBy?: string
  sortDirection?: 'ASC' | 'DESC'
}

export interface DosageResponse {
  items: Dosage[]
  totalCount: number
} 